Algorithm/src/BatchRun.cpp

```diff
@@ -0,0 +1,163 @@
+        tokens.push_back(token);
+    }
+    return tokens;
+}
+
+std::vector<std::vector<std::string>> readFile(ParamsCarrier *singleTone){
+    std::ifstream file("../Algorithm/src/geliosphere_batch_input_paramaters.csv");
+
+    // Check if the file is open
+    if (!file.is_open()) {
+        std::cerr << "Error opening the file." << std::endl;
+        return std::vector<std::vector<std::string>>();;
+    }
+
+    // Vector of vectors to store the loaded integers
+    std::vector<std::vector<std::string>> data;
+    
+    // Read and process each line
+    std::string line;
+    std::getline(file, line);
+    while (std::getline(file, line)) {
+        // Split the line into fields
+        std::vector<std::string> fields = splitString(line, ',');
+
+        std::vector<std::string> row;
+        // Process each field
+        for (const std::string &field : fields) {
+            row.push_back(field);
+            // Do something with the field
+            //std::cout << field << " ";
+        }
+        data.push_back(row);
+        // Move to the next line
+        //std::cout << std::endl;
+    }
+
+    // Close the file
+    file.close();
+
+    // Print the loaded 2D array
+    /*std::cout << "Loaded 2D array:" << std::endl;
+    for (const auto &row : data) {
+        for (std::string value : row) {
+            std::cout << value << " ";
+        }
+        std::cout << std::endl;
+    }
+    */
+    return data;
+
+
+}
+
+void generateTomlFile(double r, double theta) {
+    //std::cout << "r: " << r << " theta: " << theta << std::endl;
+    auto tbl = toml::table{
+        { "default_values", toml::table{
+            { "uniform_energy_injection_maximum", 3.0 } ,
+            { "K0", 5e+22 },
+            { "V", 400.0 },
+            { "dt", 50.0 },
+            { "r_injection", r }} 
+        },
+        { "2d_models_common_settings", toml::table{
+            { "theta_injection", theta } ,
+            { "use_uniform_injection", true }}
+        },
+        { "SolarProp_like_model_settings", toml::table{
+            { "SolarProp_ratio", 0.02 } } 
+        },
+        { "Geliosphere_model_settings", toml::table{
+            { "Geliosphere_ratio", 0.2 } ,
+            { "K0_ratio", 1.0 },
+            { "C_delta", 8.7e-05 },
+            { "default_tilt_angle", 0.1 }} 
+        },
+        { "advanced_settings", toml::table{
+            { "remove_log_files_after_simulation", true } } 
+        }
+    };
+    //std::cout << toml::toml_formatter{ tbl } << "\n";
+    //std::cout << tbl << "\n";
+    // Save to TOML file
+    //std::remove("./Settings_batch.toml");
+    //std::cout << "Snazim sa zapisat do Toml settings" << std::endl;
+    std::ofstream file("./Settings_batch.toml");
+    
+    if (file.is_open()) {
+        file << tbl;
+        file.close();
+    } else {
+        std::cerr << "Error opening file for writing." << std::endl;
+    }
+    
+}
+
+void BatchRun::runAlgorithm(ParamsCarrier *singleTone)
```

Comment:
I think we can take advantage of this and do something more generalized. We should probably use a different structure of input CSV file. It should consists of all input parameters for Geliosphere(model, year, month, K0, etc.). Combination with suggestion to move validation to Utils package we might be able to handle validation on both places.